src/Stratis.Bitcoin.Features.Dns/DnsFeature.cs

```diff
@@ -98,7 +96,8 @@ public DnsFeature(IDnsServer dnsServer, IWhitelistManager whitelistManager, ILog
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.asyncLoopFactory = asyncLoopFactory;
             this.nodeLifetime = nodeLifetime;
-            this.nodeSettings = nodeSettings;
+            dnsSettings.Load(nodeSettings);
+            this.dnsSettings = dnsSettings;
```

Comment:
@quantumagi I've thought about this a bit and its not entirely good practice to have logic in the constructor of a class. We don't want to be forced to load the DNS settings on construction. What if we want to defer that logic in the future? I had a look at your code and I'm of the opinion that the two lines above:

```
dnsSettings.Load(nodeSettings);
this.dnsSettings = dnsSettings;
```

Should in actual fact be moved to the initialize method below. What do you think?

Reply:
Ok, I removed that method.